sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -1157,6 +1183,33 @@ public virtual Response<ExtractLinkedEntitiesResultCollection> ExtractEntityLink
         #endregion
 
         #region Common
+
+        private static Dictionary<string, int> CreateIdToIndexMap(IEnumerable<TextDocumentInput> inputs)
+        {
+            var map = new Dictionary<string, int>();
+
+            int i = 0;
+            foreach (var item in inputs)
+            {
+                map[item.Id] = i++;
+            }
+
+            return map;
+        }
+
+        private static Dictionary<string, int> CreateIdToIndexMap(IEnumerable<DetectLanguageInput> inputs)
+        {
+            var map = new Dictionary<string, int>();
+
+            int i = 0;
+            foreach (var item in inputs)
+            {
+                map[item.Id] = i++;
+            }
+
+            return map;
+        }
+
         private List<TextDocumentInput> ConvertToDocumentInputs(IEnumerable<string> inputs, string language)
```

Comment:
Would be great to have a version of this method for a single input:
```cs
ConvertToDocumentInput(string input, string language, int id = 0)
    => new TextDocumentInput($"{id}", input) { Language = language ?? _options.DefaultLanguage };
```

Then `ConvertToDocumentInputs` can be simplified:
```cs
ConvertToDocumentInputs(IEnumerable<string> inputs, string language)
    => inputs.Select((input, i) => ConvertToDocumentInput(input, language, i)).ToList();
```